### PR TITLE
Fullbleed and safe zones in AMP output

### LINKED ISSUE #1245: Fullbleed and safe zones in the output
For context see: https://github.com/google/web-stories-wp/issues/1244

Prototype is https://jsbin.com/yobiyus/edit?html,css,output

Blocked on https://github.com/ampproject/amphtml/pull/27789

### PR DESCRIPTION
**Pull request by dvoytenko**
Fixes #1245

### CODE REVIEW

**Comment by wassgha** [comment]
This will clip the overlay beyond the safezone, screenshot:
Screen Shot 2020-06-04 at 5 15 00 PM

File: assets/src/edit-story/output/page.js [hunk 1]
```diff
@@ -41,7 +41,10 @@ function OutputPage({ page, autoAdvance, defaultPageDuration }) {
     backgroundSize: '20px 20px',
     backgroundPosition: '0 0, 0 10px, 10px -10px, -10px 0px',
   };
-  const backgroundOverlayStyles = generateOverlayStyles(backgroundOverlay);
+  const backgroundOverlayStyles =
```
Comment (wassgha) [inline assets/src/edit-story/output/page.js]: I'd keep this as it used to be and do the `backgroundOverlay !== OverlayType.NONE` test later

File: assets/src/edit-story/output/page.js [hunk 2]
```diff
@@ -55,30 +58,32 @@ function OutputPage({ page, autoAdvance, defaultPageDuration }) {
...
-        </amp-story-grid-layer>
-      )}
-
-      {backgroundOverlay && backgroundOverlay !== OverlayType.NONE && (
-        <amp-story-grid-layer template="vertical">
-          <div
-            className="page-background-overlay-area"
-            style={{ ...backgroundOverlayStyles }}
-          />
+      {(backgroundElements.length > 0 || backgroundOverlayStyles) && (
```
Comment (wassgha) [inline assets/src/edit-story/output/page.js]: I don't think there's a case where you have `backgroundOverlayStyles` but no `backgroundElements` no?

File: assets/src/edit-story/output/page.js [hunk 3]
```diff
@@ -55,30 +58,32 @@ function OutputPage({ page, autoAdvance, defaultPageDuration }) {
...
+          {backgroundElements.length > 0 && (
+            <div className="page-fullbleed-area" style={backgroundStyles}>
+              <div className="page-safe-area">
+                {backgroundElements.map((element) => (
+                  <OutputElement key={'el-' + element.id} element={element} />
+                ))}
+              </div>
+            </div>
+          )}
+          {backgroundOverlayStyles && (
```
Comment (wassgha) [inline assets/src/edit-story/output/page.js]:
```suggestion
          {backgroundOverlay !== OverlayType.NONE && (
```

**Comment by dvoytenko** [comment]
@wassgha 
> This will clip the overlay beyond the safezone, screenshot:

Fixed.

**Review by wassgha — APPROVED** [review]

File: assets/src/edit-story/output/page.js (hunk 2 above)
Comment (dvoytenko) [reply assets/src/edit-story/output/page.js]: Not sure, but the old code seemed to have thought so. Can you confirm?

**Comment by dvoytenko** [comment]
@wassgha all fixed. ptal.

**Pull request merged**